### PR TITLE
add a test

### LINKED ISSUE #64: claims with arrays cause a stringified payload to be created, rather than an object
Note below that roles and permissions are arrays. If either has more than one element, the entire payload becomes stringified:

```
// Run this and paste the token into the debugger at jwt.io, you will observe a stringified payload

auth_token = auth.GetToken({
    return {
        nickname: "jest.jester",
        name: "jest@itest.com",
        updated_at: "2022-10-09T11:36:37.582Z",
        email: "jest@itest.com",
        iss: `${domain}/`,
        sub: "auth0|633c191bd579670011607e98",
        aud: "TWqR7DTIezpfW82qXFt0lchMXixIpFLQ",
        iat: 1665315399,
        exp: 5665351399,
        sid: "lBZb4EUN7LpXu7Urp-lRQ3qnCzDVBf23",
        nonce: "RzZtVlVTT1dqZ1lvWFpSYzFnbnB2RXZkdWcuNDJQRkREUH5KMUVJNWdMSA==",
        permissions: ["read_all"],
        roles: ["admin", "leader"]
    };
}
```

When only one element appears in the arrays, all is well

```
// Run this and paste the token into the debugger at jwt.io, you will observe a stringified payload

auth_token = auth.GetToken({
    return {
        nickname: "jest.jester",
        name: "jest@itest.com",
        updated_at: "2022-10-09T11:36:37.582Z",
        email: "jest@itest.com",
        iss: `${domain}/`,
        sub: "auth0|633c191bd579670011607e98",
        aud: "TWqR7DTIezpfW82qXFt0lchMXixIpFLQ",
        iat: 1665315399,
        exp: 5665351399,
        sid: "lBZb4EUN7LpXu7Urp-lRQ3qnCzDVBf23",
        nonce: "RzZtVlVTT1dqZ1lvWFpSYzFnbnB2RXZkdWcuNDJQRkREUH5KMUVJNWdMSA==",
        permissions: ["read_all"],
        roles: ["admin"]
    };
}
```

** JWT.io **

image

image

### PR DESCRIPTION
**Pull request by levino**
this adds a test which is always good

### CODE REVIEW

File: src/tools.ts [hunk 1]
```diff
@@ -144,7 +144,7 @@ export const createKeyPair = () => {
   }
 }
 
-export interface JwtPayload {
+export interface JwtPayload extends Record<string, unknown> {
```
Comment (levino) [inline src/tools.ts:147]: Might be useful for people add custom props to the payload. I am not sure whether this is legal...

**Comment by chrisnurse** [comment]
Not sure if it is useful to paste this here, but if you need an input to the test that I used...

  const claims = {
      nickname: "jest.jester",
      name: "jest@itest.com",
      updated_at: "2022-10-09T11:36:37.582Z",
      email: "jest@itest.com",
      iss: "https://jest.au.auth0.com/",
      sub: "auth0|633c191bd579670011607e98",
      aud: "TWqR7DTIezpfW82qXFt0lchMXixIpFLQ",
      iat: 1665315399,
      exp: 5665351399,
      sid: "lBZb4EUN7LpXu7Urp-lRQ3qnCzDVBf23",
      nonce: "RzZtVlVTT1dqZ1lvWFpSYzFnbnB2RXZkdWcuNDJQRkREUH5KMUVJNWdMSA==",
      roles: [
        "admin",
      ],
      permissions: [
        "create:all",
        "read:all",
        "write:all",
        "update:all",
        "delete:all",
      ],
    }

Bear in mind it's not an issue of mock-jwks decoding tokens that it has encoded. The issue seems to be more compatibility with other libraries and the fact that JWT.io debugger shows a stringified payload when decoding a token created with the above input.

**Comment by levino** [comment]
I [now log the token](https://app.circleci.com/pipelines/github/levino/mock-jwks/120/workflows/f58ec481-71de-44b5-bae2-1aa0f5e65065/jobs/334?invite=true#step-103-9). When I paste it to jwt.io, all is fine. So I still cannot reproduce. Please let me know if I am missing something. Also please feel free to fork this repo and play around with the branch to help reproduce the issue.

**Comment by levino** [comment]
Another [attempt](https://app.circleci.com/pipelines/github/levino/mock-jwks/121/workflows/ee76580c-943d-4cc9-83cc-4c8f6bc8b3e1/jobs/338?invite=true#step-103-9) by calling `token` on the `jwksMock` object. I still get a nice payload from jwt.io when I paste the token from the ci logs.

**Comment by chrisnurse** [comment]
Thanks, I think you are on the right track.

When I manually edit the tools.js, I can fix the issue by removing the "buffering" step as indicated in these screen clips...

image

If tools.ts transforms to this js, then you've nailed it. I'll see if I can pursue further.

**Comment by chrisnurse** [comment]
Looking at tools.ts I see where the stringification occurs. If I remove this in the JS, just so I can hack the package to repeat my steps, all appears fine. I am not sure if there are other side-effects of this change of course.

image

**Comment by levino** [comment]
As I said above. When I log the string returned from
```typescript
auth0Mock.token({
  nickname: 'jest.jester',
  name: 'jest@itest.com',
  updated_at: '2022-10-09T11:36:37.582Z',
  email: 'jest@itest.com',
  iss: 'https://jest.au.auth0.com/',
  sub: 'auth0|633c191bd579670011607e98',
  aud: 'TWqR7DTIezpfW82qXFt0lchMXixIpFLQ',
  iat: '1665315399',
  exp: '5665351399',
  sid: 'lBZb4EUN7LpXu7Urp-lRQ3qnCzDVBf23',
  nonce: 'RzZtVlVTT1dqZ1lvWFpSYzFnbnB2RXZkdWcuNDJQRkREUH5KMUVJNWdMSA==',
  roles: ['admin'],
  permissions: [
    'create:all',
    'read:all',
    'write:all',
    'update:all',
    'delete:all',
  ],
})
```
I get
```
eyJhbGciOiJSUzI1NiIsImtpZCI6IlYwUk9SbXczVlU5elkzUnZTMjFtU2xRcmVUWnJRMUZKWWt4SlBRIn0.eyJuaWNrbmFtZSI6Implc3QuamVzdGVyIiwibmFtZSI6Implc3RAaXRlc3QuY29tIiwidXBkYXRlZF9hdCI6IjIwMjItMTAtMDlUMTE6MzY6MzcuNTgyWiIsImVtYWlsIjoiamVzdEBpdGVzdC5jb20iLCJpc3MiOiJodHRwczovL2plc3QuYXUuYXV0aDAuY29tLyIsInN1YiI6ImF1dGgwfDYzM2MxOTFiZDU3OTY3MDAxMTYwN2U5OCIsImF1ZCI6IlRXcVI3RFRJZXpwZlc4MnFYRnQwbGNoTVhpeElwRkxRIiwiaWF0IjoiMTY2NTMxNTM5OSIsImV4cCI6IjU2NjUzNTEzOTkiLCJzaWQiOiJsQlpiNEVVTjdMcFh1N1VycC1sUlEzcW5DekRWQmYyMyIsIm5vbmNlIjoiUnpadFZsVlRUMWRxWjFsdldGcFNZekZuYm5CMlJYWmtkV2N1TkRKUVJrUkVVSDVLTVVWSk5XZE1TQT09Iiwicm9sZXMiOlsiYWRtaW4iXSwicGVybWlzc2lvbnMiOlsiY3JlYXRlOmFsbCIsInJlYWQ6YWxsIiwid3JpdGU6YWxsIiwidXBkYXRlOmFsbCIsImRlbGV0ZTphbGwiXX0.SXWwBsRdqIsnOk8XGsarbXjHh9l_-bQhF7rMVZqxsB2XQ3UaXr5ZJR63vFUSyN02nfPKNJYeWM2KL97OUGeg2a3dDXXyYp8JO2tr_KC2_M34q_dAHxRpSE26WPTnGzJQlAiSR_PZ739VFo6fgDC94YKjoBxIoMxFM_KK9d7nm7OS6DLf3fdP3MHaPoH_wZPV2-B9wJHGg4NukaGMC7TzHNcF5-VAciGuKYhGq8EbloHDHiA-drzMORQJlMyFzJt0z92Je-ggRU_wca-gzoNF1PeFBAMBeWVeZ5hcZgXrpx0kmIX3RO2_x_nwvxc-t0nEVYkwu6J71C60Y2Njritz1A
```
as you can see in the [ci builds logs](url). When I copy above token to jwt.io, I get a proper result:
Screenshot 2022-10-17 at 09 38 18

Please provide a code snippet that produces an invalid token. Not screenshots please. Maybe you can use codesandbox.io to produce the token?

**Comment by levino** [comment]
Also maybe it is a problem with your `Buffer` or `JSON` object? Which node version are you running?

**Comment by levino** [comment]
Also please focus on the reproduction of the issue, not the solution. I will not merge any "fixes" without being able to reproduce the problem first.

**Pull request merged**